### PR TITLE
Add property as IPC interface capability.

### PR DESCRIPTION
**Pull request by mamokarz**

### CODE REVIEW

File: inc/az_ulib_ipc_api.h
Comment (mamokarz) [inline inc/az_ulib_ipc_api.h:411]: >command_index [](http://example.com/codeflow?start=29&length=13)

property_index

File: inc/az_ulib_ipc_api.h
Comment (mamokarz) [inline inc/az_ulib_ipc_api.h:459]: >command_index [](http://example.com/codeflow?start=29&length=13)

property_index

**Comment by smcd253** [comment]
lgtm. make sure the command_index/property_index problem on clang and osx is addressed before merging.

File: inc/az_ulib_descriptor_api.h
Comment (cindydeng1998) [inline inc/az_ulib_descriptor_api.h:236]: nit: align braces on same line like in AZ_ULIB_DESCRIPTOR_ADD_COMMAND above, or give them separate lines in the two above

Comment (mamokarz) [reply inc/az_ulib_descriptor_api.h:236]: This alignend is defined by the clang-format as required by our code guidelines.

Comment (cindydeng1998) [reply inc/az_ulib_descriptor_api.h:236]: so it has to be different within the same file? see line 171 and 200 compared to 236/237

Comment (mamokarz) [reply inc/az_ulib_descriptor_api.h:236]: Yep, clang-format decides the "best" way. I've just sent you an email how to enable it in VSCode.

**Review by danewalton — APPROVED** [review]

File: inc/az_ulib_ipc_api.h
Comment (smcd253) [inline inc/az_ulib_ipc_api.h:394]: clang throws an error here. must be command_index.

**Review by smcd253 — APPROVED** [review]
lgtm

File: inc/az_ulib_ipc_api.h
Comment (mamokarz) [inline inc/az_ulib_ipc_api.h:383]: >az_ulib_ipc_get [](http://example.com/codeflow?start=23&length=15)

After some discussions offline, we decided that we don't need get and set APIs. So I'll remove those.

File: tests/inc/az_ulib_test_my_interface.h
Comment (smcd253) [inline tests/inc/az_ulib_test_my_interface.h:101]: I remember you saying that you were going to hold off on async capability until later. I see it was included earlier, is it still possible to use async commands with the generic capability interface?

**Review by smcd253 — APPROVED** [review]
lgtm

File: tests/inc/az_ulib_test_my_interface.h
Comment (mamokarz) [reply tests/inc/az_ulib_test_my_interface.h:101]: I'll put it back whene we decide to implent it for real.

**Review by danewalton — APPROVED** [review]

**Review by cindydeng1998 — APPROVED** [review]
🌞

File: inc/az_ulib_ipc_interface.h
Comment (mamokarz) [reply inc/az_ulib_ipc_interface.h:78]: It was a feedback from Jeffrey, vtable has a very specific meaning and it is not totally correct here.

**Pull request merged**
